### PR TITLE
feat: finish msg inclusion api

### LINKED ISSUE #381: Create API to check for message inclusion
## Summary

Create an API to check for messages inclusion.

## Details

The first step has already been done by writing code to find subtree root paths. The next step should be to create an API that allows us to easily check for message inclusion given the original data square, the position of the message in question, the size of that message, along with the commitment provided in the `SignedPayForMessage`.

### Action Items
- [ ] define API for checking inclusion
- [ ] compare commitments and subtree roots the `SignedPayForData` and the actual data in the square given the location and message size.
- [ ] write integration tests for subtree root code

### References

depends on [#49](https://github.com/celestiaorg/nmt/pull/49)

### PR DESCRIPTION
**Pull request by evan-forbes**
finishes the api for message inclusion 

part of #382 
closes #381

### CODE REVIEW

File: pkg/inclusion/paths_test.go [hunk 1]
```diff
@@ -232,18 +245,65 @@ func Test_genSubTreeRootPath(t *testing.T) {
...
+func Test_calculateCommitPaths(t *testing.T) {
+	type test struct {
+		size, start, msgLen int
+		expected            []path
+	}
+	tests := []test{
+		{2, 0, 1, []path{{instructions: []WalkInstruction{WalkLeft}, row: 0}}},
+		{2, 2, 2, []path{{instructions: []WalkInstruction{}, row: 1}}},
+		{2, 1, 2, []path{{instructions: []WalkInstruction{}, row: 1}}},
+		{4, 2, 2, []path{{instructions: []WalkInstruction{WalkRight}, row: 0}}},
```
Comment (rootulp) [inline pkg/inclusion/paths_test.go:273]: [no change needed] visual

```ascii
row 0 -> | | |2|2|
row 2 -> | | | | |
row 3 -> | | | | |
row 4 -> | | | | |
```

merkle tree of row 0
```ascii
        x
       / \
      x   y
         / \
        2   2
```
where `y` is what we want

Comment (evan-forbes) [reply pkg/inclusion/paths_test.go:273]: exactly!

File: pkg/inclusion/paths.go [hunk 2]
```diff
@@ -1,17 +1,61 @@
...
+		}
+	}
+
+	return paths
+}
+
 // genSubTreeRootPath calculates the path to a given subtree root of a node, given the
 // depth and position of the node. note: the root of the tree is depth 0.
 // The following nolint can be removed after this function is used.
 //nolint:unused,deadcode
```
Comment (SweeXordious) [inline pkg/inclusion/paths.go:50]: [Non Blocking][Suggestion]

Probably can be removed as this function is now called.
```suggestion
```

**Review by SweeXordious — APPROVED** [review]
:rocket:

File: pkg/inclusion/paths.go (hunk 2 above)
Comment (evan-forbes) [reply pkg/inclusion/paths.go:50]: nice! good find, I'll if there is other feedback, I'll include this in this PR, if not, I'll be sure to include this in the next `ProcessProposal` PR as to not dismiss reviews.

**Pull request merged**